pkg/credentialprovider/config.go

```diff
@@ -287,8 +287,8 @@ func decodeDockerConfigFieldAuth(field string) (username, password string, err e
 
 	// StdEncoding can only decode padded string
 	// RawStdEncoding can only decode unpadded string
-	// a string is correctly padded if and only if its length is a multiple of 4
-	if (len(field) % 4) == 0 {
+	field = strings.TrimRight(field, "\r\n")
+	if field != "" && field[len(field)-1] == '=' {
```

Comment:
Could we please use a descriptive variable name to make it clear that `field[len(field)-1] == '=' is a check of whether the string is padded?

Thanks :)